api/app/Extensions/JWTAuth/JWTAuth.php

```diff
@@ -0,0 +1,58 @@
+<?php namespace App\Extensions\JWTAuth;
+
+use Illuminate\Http\Request;
+use Tymon\JWTAuth\Token;
+use Tymon\JWTAuth\JWTAuth as JWTAuthBase;
+use RuntimeException;
+use App\Exceptions\BadRequestException;
+use App\Exceptions\UnauthorizedException;
+use App\Exceptions\TokenInvalidException;
+use App\Exceptions\UnprocessableEntityException;
+use Tymon\JWTAuth\Exceptions\TokenExpiredException;
+
+class JWTAuth extends JWTAuthBase
+{
+    /**
+     * Get the token contained within the current request.
+     *
+     * @param  Request  $request
+     * @throws BadRequestException
+     * @return Token
+     */
+    public function getTokenFromRequest(Request $request)
```

Comment:
Yes, the request object contained from the constructor is quite empty. I'm looking into if we can change that behaviour and remove the need to set the request with the setRequest() method.